src/legacy/core_plugins/telemetry/server/telemetry_repository/update_telemetry_saved_object.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { TelemetrySavedObjectAttributes } from './';
+import { SavedObjectsErrorHelpers } from '../../../../../core/server';
+
+export async function updateTelemetrySavedObject(
+  savedObjectsClient: any,
+  savedObjectAttributes: TelemetrySavedObjectAttributes
+) {
+  try {
+    return await savedObjectsClient.update('telemetry', 'telemetry', savedObjectAttributes);
+  } catch (err) {
+    if (SavedObjectsErrorHelpers.isNotFoundError(err)) {
+      const savedObjectAttributesWithDefaults = {
+        enabled: null,
+        ...savedObjectAttributes,
+      };
+      return await savedObjectsClient.create('telemetry', savedObjectAttributesWithDefaults, {
+        id: 'telemetry',
+        overwrite: true,
+      });
+    }
```

Comment:
I'm a bit uneasy about automatically creating an object when an update fails.

I think the reasons it makes me uncomfortable is that:
1. This could hide a logical error where we keep trying to update the wrong object (if, for example, you use the wrong _id_ field) and we wouldn't find out until lots of incorrect objects are created in Elasticsearch.
2. We're doing a _dangerous_ operation in the catch of another operation (which failed)... this feels wrong.

Do we have a good reason for trying to update a non existant object?